src/lib/util/Notifications/Twitch.ts

```diff
@@ -0,0 +1,91 @@
+import * as crypto from 'crypto';
+import { TOKENS } from '../../../../config';
+import { fetch, enumerable, FetchResultTypes } from '../util';
+import { Mime } from '../constants';
+import { TwitchKrakenChannelSearchResults, TwitchHelixResponse, TwitchHelixGameSearchResult } from '../../types/definitions/Twitch';
+import { RateLimitManager } from 'klasa';
+
+const enum ApiVersion {
+	Kraken,
+	Helix
+}
+
+export class Twitch {
+
+	public ratelimitsStreams: RateLimitManager = new RateLimitManager(1, Twitch.RATELIMIT_COOLDOWN);
+
+	@enumerable(false)
+	private readonly $clientID: string = TOKENS.TWITCH.CLIENT_ID;
+
+	@enumerable(false)
+	private readonly $webhookSecret: string = TOKENS.TWITCH.WEBHOOK_SECRET;
+
+	private readonly BASE_URL_HELIX: string = 'https://api.twitch.tv/helix/';
+	private readonly BASE_URL_KRAKEN: string = 'https://api.twitch.tv/kraken/';
+
+	private kFetchOptions = {
+		headers: {
+			'Accept': Mime.Types.ApplicationTwitchV5Json,
+			'Client-ID': this.$clientID
+		}
+	} as const;
+
+	public streamNotificationLimited(id: string) {
+		const existing = this.ratelimitsStreams.get(id);
+		return existing && existing.limited;
+	}
+
+	public streamNotificationDrip(id: string) {
+		try {
+			this.ratelimitsStreams.acquire(id).drip();
+		} catch { return true; }
+		return false;
+	}
+
+	public async fetchUsersByLogin(logins: string[]): Promise<TwitchKrakenChannelSearchResults> {
+		return this._performApiGETRequest(`users?login=${this._formatMultiEntries(logins, true)}`);
+	}
+
+	public async fetchGame(ids: string[] = [], names: string[] = []): Promise<TwitchHelixResponse<TwitchHelixGameSearchResult>> {
+		let search = '';
+		for (const id of ids) search = `${search}&id=${encodeURIComponent(id)}`;
+		for (const name of names) search = `${search}&name=${encodeURIComponent(name)}`;
+		return this._performApiGETRequest(`games?${search}`, ApiVersion.Helix);
+	}
+
+	private _formatMultiEntries(data: string[], replaceEncode?: boolean) {
+		let out = data.map(encodeURIComponent).join(',');
+		replaceEncode ? out = out.replace(/%20/g, '&20') : null;
+		return out;
+	}
+
+	private async _performApiGETRequest<T>(path: string, api: ApiVersion = ApiVersion.Kraken): Promise<T> {
+		const result = await fetch(`${api === ApiVersion.Kraken ? this.BASE_URL_KRAKEN : this.BASE_URL_HELIX}${path}`, this.kFetchOptions, FetchResultTypes.JSON) as unknown as T;
+		return result;
+	}
+
+	public static RATELIMIT_COOLDOWN: number = 10 * 60 * 1000;
```

Comment:
This should use the Time enum found in `lib/util/constants.ts` and multiply `Time.Minute` by 10

Reply:
Resolved with 38cb96f.